tests/dflowfm/test_net.py

```diff
@@ -765,3 +766,62 @@ def test_network_is_geographic():
 
     network = Network(is_geographic=True)
     assert network.is_geographic == True
+
+
+def check_ugrid_consistency(ds: xr.Dataset):
+    # inspired by xugrid.ugrid.conventions._get_coordinates:
+    # https://github.com/Deltares/xugrid/blob/412d9ae0b01d97f3eadabc81a130f9d53f5a1dcb/xugrid/ugrid/conventions.py#L221-L247
+    _COORD_NAMES = {
+        1: ("node_coordinates", "edge_coordinates"),
+        2: ("node_coordinates", "face_coordinates", "edge_coordinates"),
+    }
+    topologies = [k for k in ds.data_vars if ds[k].attrs.get("cf_role") == "mesh_topology"]
+    
+    for topology in topologies:
+        attrs = ds[topology].attrs
+        topodim = attrs["topology_dimension"]
+        for name in _COORD_NAMES[topodim]:
+            if name in attrs:
+                candidates = [c for c in attrs[name].split(" ") if c in ds]
+                if len(candidates) == 0:
+                    raise KeyError(
+                        f"the following variables are specified for UGRID {name}: "
+                        f'"{attrs[name]}", but they are not present in the dataset'
+                    )
+
+
+def test_network_netcdf_consistency(tmp_path):
+    from hydrolib.core.dflowfm import NetworkModel
+    import xarray as xr
```

Comment:
xarray is already imported at the top of the file? You can also add the import of the NetworkModel at the top of the test file (if it is not there yet)